src/store/thunks/companies.ts

```diff
@@ -0,0 +1,12 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+
+import companies from "@app/data/companies.json";
+import { Company } from "@app/interfaces";
+
+export const fetchCompanies = createAsyncThunk<Company[], string[]>(
+  "company/fetchCompanies",
+  async (ids) => {
+    await new Promise((resolve) => setTimeout(resolve, 300));
+    return companies.filter((company) => ids.includes(company.id));
```

Comment:
Цикл в цикле. Лучше такого избегать. Используй Map и\или Set для оптимизации поиска (вместо includes)

```suggestion
    const idsSet = new Set(ids);
    return companies.filter((company) => idsSet.has(company.id));
```